velox/expression/tests/ExprTest.cpp

```diff
@@ -4942,6 +4942,33 @@ TEST_F(ExprTest, disableMemoization) {
   ASSERT_EQ(stats["plus"].numProcessedRows, 3 * flatSize);
 }
 
+TEST_F(
+    ExprTest,
+    verifyOuterDictionaryLayersInitalizedWhenLoggingLazyLoadedVector) {
+  // Verify that outer dictionary layers are initialized when performing a
+  // simple vector logging.
+  auto doubleVector = makeFlatVector<double>({11.0, 12.0, 13.0});
+  auto doubleInDictionary = BaseVector::wrapInDictionary(
+      nullptr, makeIndices({0, 1, 2}), 3, doubleVector);
+  auto doubleInNestedDictionary = BaseVector::wrapInDictionary(
+      nullptr, makeIndices({0, 1, 2}), 3, doubleInDictionary);
+  auto doubleInLazyDictionary =
+      VectorFuzzer::wrapInLazyVector(doubleInNestedDictionary);
+
+  auto rowVector = makeRowVector({doubleVector, doubleInLazyDictionary});
+
+  // log vector
+  for (const auto& child : rowVector->children()) {
+    LOG(INFO) << child->toString(/*recursive=*/true) << std::endl;
+  }
+  for (vector_size_t i = 0; i < rowVector->size(); ++i) {
+    LOG(INFO) << rowVector->toString(i) << std::endl;
+  }
```

Comment:
I copied this code from an existing utility method, so it calls toString of both child and rowVector. In your unit test, you probably only need one of these loops to trigger the error.